core_v14/src/metadata/mod.rs

```diff
@@ -26,12 +26,12 @@ use frame_metadata::{RuntimeMetadata, RuntimeMetadataPrefixed};
 use scale_info::{form::PortableForm, PortableRegistry};
 use std::collections::HashMap;
 
-// We don't expose anything scale-info or parity-scale-codec related outside of
-// this crate currently, so no need to expose these either:
-pub(crate) type Type = scale_info::Type<PortableForm>;
-pub(crate) type TypeDef = scale_info::TypeDef<PortableForm>;
-pub(crate) type TypeId = <scale_info::form::PortableForm as scale_info::form::Form>::Type;
-pub(crate) type SignedExtensionMetadata = frame_metadata::SignedExtensionMetadata<PortableForm>;
+// Some type aliases used below. `scale-info` is re-exported at the root,
+// so to avoid confusion we only publically export all scale-info types from that
+// one place.
+type TypeId = <scale_info::form::PortableForm as scale_info::form::Form>::Type;
```

Comment:
@TarikGul to expound on the above, here James is *not* using `pub type TypeId` because it's already re-exported above and having the same type available through separate paths is annoying (but certainly possible, e.g. `core::time::Duration` and `std::time::Duration` are both available interchangeably).